R/process_lookup_homelessness.R

```diff
@@ -0,0 +1,99 @@
+#' create homelessness lookup
+#' @description reads in homelessness extract to create flags
+#'
+#' @param year The year to process, in FY format.
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#' @family process extracts
+#'
+create_homelessness_lookup <- function(
+    year,
+    homelessness_data = read_file(get_source_extract_path(year, "Homelessness"))) {
+  homelessness_lookup <- homelessness_data %>%
+    dplyr::distinct(.data$chi, .data$record_keydate1, .data$record_keydate2) %>%
+    tidyr::drop_na(.data$chi) %>%
+    dplyr::mutate(hl1_in_fy = 1L)
+
+  return(homelessness_lookup)
+}
+
+
+#' add homelessness flag episode
+#' @description add homelessness in FY flag to episode/individual file
+#'
+#' @param data The extract to process
+#' @param year The year to process, in FY format.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package]
+#' @export
```

Comment:
We possibly want to rename this function.

Some doc tweaks anyway:
```suggestion
#' Add 'homelessness in FY' flag
#' @description Add a flag to the data indicating if the CHI
#' had a homelessness episode within the financial year.
#'
#' @param data The data to add the flag to - the episode 
#' or individual file.
#' @param lookup The homelessness lookup created by [create_homelessness_lookup()]
#' @inheritParams run_episode_file
#'
#' @return the final data as a [tibble][tibble::tibble-package]
#' @export
```